src/FakeItEasy/Configuration/ArgumentCollection.cs

```diff
@@ -54,13 +51,13 @@ internal ArgumentCollection(object[] arguments, MethodInfo method)
         public int Count
         {
             [DebuggerStepThrough]
-            get { return this.arguments.Length; }
+            get => this.arguments.Length;
         }
 
         /// <summary>
         ///   Gets the names of the arguments in the list.
         /// </summary>
-        public IEnumerable<string> ArgumentNames { get; }
+        public IEnumerable<string> ArgumentNames => this.Method.GetParameters().Select(x => x.Name);

```

Comment:
The problem with this is that it re-evaluates `GetParameters().Select(...)` every time it's called. It's used in `Get<T>(string argumentName)`, which is used in various scenarios (e.g. `Where`, `WhenArgumentsMatch`, `Invokes`, etc). So it could reduce performance in those scenarios. If it's really important to evaluate it on demand, it could be done with a `Lazy<T>`.

Reply:
It's true. I was assuming `Get<T>(string)` was used infrequently. Happy to change to lazily calculated and stored.